zebra-chain/src/orchard/tree.rs

```diff
@@ -292,6 +295,71 @@ pub struct NoteCommitmentTree {
     cached_root: std::sync::RwLock<Option<Root>>,
 }
 
+/// The block height of the final leaf in a level 16 subtree and
+/// the block height of the first leaf of the next subtree
+#[derive(Clone, Debug)]
```

Comment:
Nit: Please derive all useful traits on new structs.

I'm not sure if `PartialEq` should compare the entire struct, or just the `index` (or just `index` and `root`). But if we're mainly going to use it in tests, maybe it should compare all the fields.

(Maybe it would help to create a self-review checklist for these kinds of things?)

```suggestion
#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```